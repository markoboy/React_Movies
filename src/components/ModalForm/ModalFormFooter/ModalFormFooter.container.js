import Button from '@components/common/Forms/Button';
import Column from '@components/common/Grid/Column';
import Row from '@components/common/Grid/Row';
import {
  MODAL_FORM_ADD_ACTION,
  MODAL_FORM_EDIT_ACTION,
} from '@constants/Modal';
import React, { memo, useMemo } from 'react';

function ModalFormFooterContainer({ action }) {
  const { hasResetButton, submitActionText } = useMemo(() => {
    let actionText;

    const resetButton = [
      MODAL_FORM_EDIT_ACTION,
      MODAL_FORM_ADD_ACTION,
    ].includes(action);

    switch (action) {
      case MODAL_FORM_EDIT_ACTION:
        actionText = 'Save';
        break;

      case MODAL_FORM_ADD_ACTION:
        actionText = 'Submit';
        break;

      default:
        actionText = 'Confirm';
        break;
    }

    return {
      hasResetButton: resetButton,
      submitActionText: actionText,
    };
  }, [action]);

  return (
    <Row classes="justify-content--flex-end">
      {hasResetButton && (
        <Column classes="column--m-3">
          <Button type="reset" classes="btn--outline btn--full-width">
            Reset
          </Button>
        </Column>
      )}

      <Column classes="column--m-3">
        <Button type="submit" classes="btn--primary btn--full-width">
          {submitActionText}
        </Button>
      </Column>
    </Row>
  );
}

export default memo(ModalFormFooterContainer);
