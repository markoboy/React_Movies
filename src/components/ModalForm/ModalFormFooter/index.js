export { default } from './ModalFormFooter.container';
