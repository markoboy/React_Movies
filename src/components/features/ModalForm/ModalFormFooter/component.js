import Button from '@components/common/Forms/Button';
import Column from '@components/common/Grid/Column';
import Row from '@components/common/Grid/Row';
import PropTypes from 'prop-types';
import React from 'react';

function ModalFormFooterComponent({ hasResetButton, submitActionText }) {
  return (
    <Row classes="justify-content--flex-end">
      {hasResetButton && (
        <Column classes="column--m-3">
          <Button type="reset" classes="btn--outline btn--full-width">
            Reset
          </Button>
        </Column>
      )}

      <Column classes="column--m-3">
        <Button type="submit" classes="btn--primary btn--full-width">
          {submitActionText}
        </Button>
      </Column>
    </Row>
  );
}

ModalFormFooterComponent.propTypes = {
  hasResetButton: PropTypes.bool.isRequired,
  submitActionText: PropTypes.string.isRequired,
};

export default ModalFormFooterComponent;
