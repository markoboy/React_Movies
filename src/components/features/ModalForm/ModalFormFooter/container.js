import {
  MODAL_FORM_ADD_ACTION,
  MODAL_FORM_EDIT_ACTION,
} from '@components/features/ModalForm/constants';
import PropTypes from 'prop-types';
import React, { memo, useMemo } from 'react';
import ModalFormFooterComponent from './component';

function ModalFormFooterContainer({ action }) {
  const { hasResetButton, submitActionText } = useMemo(() => {
    let actionText;

    const resetButton = [
      MODAL_FORM_EDIT_ACTION,
      MODAL_FORM_ADD_ACTION,
    ].includes(action);

    switch (action) {
      case MODAL_FORM_EDIT_ACTION:
        actionText = 'Save';
        break;

      case MODAL_FORM_ADD_ACTION:
        actionText = 'Submit';
        break;

      default:
        actionText = 'Confirm';
        break;
    }

    return {
      hasResetButton: resetButton,
      submitActionText: actionText,
    };
  }, [action]);

  return (
    <ModalFormFooterComponent
      hasResetButton={hasResetButton}
      submitActionText={submitActionText}
    />
  );
}

ModalFormFooterContainer.propTypes = {
  action: PropTypes.string.isRequired,
};

export default memo(ModalFormFooterContainer);
