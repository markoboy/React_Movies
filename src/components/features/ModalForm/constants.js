export const MODAL_ADD_TITLE = 'Add Movie';
export const MODAL_EDIT_TITLE = 'Edit Movie';
export const MODAL_DELETE_TITLE = 'Delete Movie';

export const MODAL_FORM_ADD_ACTION = 'Add';
export const MODAL_FORM_EDIT_ACTION = 'Edit';
export const MODAL_FORM_DELETE_ACTION = 'Delete';

export const MODAL_FORM_DELETE_PARAGRAPH = 'Are you sure you want to delete this movie?';

export const GENRE_SELECT_OPTIONS = [
  {
    label: 'Documentary',
    value: 'documentary',
  },
  {
    label: 'Comedy',
    value: 'comedy',
  },
  {
    label: 'Horror',
    value: 'horror',
  },
  {
    label: 'Crime',
    value: 'crime',
  },
];
