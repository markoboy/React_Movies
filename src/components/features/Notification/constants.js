export const SUCCESS_TITLE = 'Congratulations!';
export const ERROR_TITLE = 'Oops an Error occurred!';
