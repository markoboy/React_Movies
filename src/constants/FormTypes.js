export const FORM_TYPE_TEXT = 'text';
export const FORM_TYPE_DATE = 'date';
export const FORM_TYPE_SELECT = 'select';
export const FORM_TYPE_NUMBER = 'number';
