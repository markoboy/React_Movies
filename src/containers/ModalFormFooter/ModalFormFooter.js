import Button from '@components/Forms/Button';
import Column from '@components/Grid/Column';
import Row from '@components/Grid/Row';
import {
  MODAL_FORM_ADD_ACTION,
  MODAL_FORM_EDIT_ACTION,
} from '@constants/Modal';
import React, { memo, useMemo } from 'react';

function ModalFormFooter({ action }) {
  const { hasResetButton, submitActionText } = useMemo(() => {
    let actionText;

    const resetButton = [
      MODAL_FORM_EDIT_ACTION,
      MODAL_FORM_ADD_ACTION,
    ].includes(action);

    switch (action) {
      case MODAL_FORM_EDIT_ACTION:
        actionText = 'Save';
        break;

      case MODAL_FORM_ADD_ACTION:
        actionText = 'Submit';
        break;

      default:
        actionText = 'Confirm';
        break;
    }

    return {
      hasResetButton: resetButton,
      submitActionText: actionText,
    };
  }, [action]);

  return (
    <Row classes="justify-content--flex-end">
      {hasResetButton && (
        <Column classes="column--m-3">
          <Button type="reset" classes="btn--outline btn--full-width">
            Reset
          </Button>
        </Column>
      )}

      <Column classes="column--m-3">
        <Button type="submit" classes="btn--primary btn--full-width">
          {submitActionText}
        </Button>
      </Column>
    </Row>
  );
}

export default memo(ModalFormFooter);
