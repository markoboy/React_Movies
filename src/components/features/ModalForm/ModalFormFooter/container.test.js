import React from 'react';
import { render } from '@testing-library/react';

import ModalFormFooter from './container';
import { MODAL_FORM_ADD_ACTION, MODAL_FORM_DELETE_ACTION, MODAL_FORM_EDIT_ACTION } from '../constants';

describe('ModalFormFooter Component', () => {
  it('renders Submit action button when action is Add', () => {
    const { container } = render(<ModalFormFooter action={MODAL_FORM_ADD_ACTION} />);

    expect(container).toMatchSnapshot();
  });

  it('renders Save action button when action is Edit', () => {
    const { container } = render(<ModalFormFooter action={MODAL_FORM_EDIT_ACTION} />);

    expect(container).toMatchSnapshot();
  });

  it('renders Confirm action button when action is Delete', () => {
    const { container } = render(<ModalFormFooter action={MODAL_FORM_DELETE_ACTION} />);

    expect(container).toMatchSnapshot();
  });

  it('renders Reset action button when action is Add', () => {
    const { container } = render(<ModalFormFooter action={MODAL_FORM_ADD_ACTION} />);

    expect(container).toMatchSnapshot();
  });

  it('renders Reset action button when action is Edit', () => {
    const { container } = render(<ModalFormFooter action={MODAL_FORM_EDIT_ACTION} />);

    expect(container).toMatchSnapshot();
  });

  it('doesnt render Reset action button when action is Delete', () => {
    const { container } = render(<ModalFormFooter action={MODAL_FORM_DELETE_ACTION} />);

    expect(container).toMatchSnapshot();
  });
});
