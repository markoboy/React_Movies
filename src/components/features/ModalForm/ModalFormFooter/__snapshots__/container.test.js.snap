// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFormFooter Component doesnt render Reset action button when action is Delete 1`] = `
<div>
  <div
    class="row justify-content--flex-end"
  >
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--primary btn--full-width"
        tabindex="0"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalFormFooter Component renders Confirm action button when action is Delete 1`] = `
<div>
  <div
    class="row justify-content--flex-end"
  >
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--primary btn--full-width"
        tabindex="0"
        type="submit"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalFormFooter Component renders Reset action button when action is Add 1`] = `
<div>
  <div
    class="row justify-content--flex-end"
  >
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--outline btn--full-width"
        tabindex="0"
        type="reset"
      >
        Reset
      </button>
    </div>
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--primary btn--full-width"
        tabindex="0"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalFormFooter Component renders Reset action button when action is Edit 1`] = `
<div>
  <div
    class="row justify-content--flex-end"
  >
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--outline btn--full-width"
        tabindex="0"
        type="reset"
      >
        Reset
      </button>
    </div>
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--primary btn--full-width"
        tabindex="0"
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalFormFooter Component renders Save action button when action is Edit 1`] = `
<div>
  <div
    class="row justify-content--flex-end"
  >
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--outline btn--full-width"
        tabindex="0"
        type="reset"
      >
        Reset
      </button>
    </div>
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--primary btn--full-width"
        tabindex="0"
        type="submit"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalFormFooter Component renders Submit action button when action is Add 1`] = `
<div>
  <div
    class="row justify-content--flex-end"
  >
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--outline btn--full-width"
        tabindex="0"
        type="reset"
      >
        Reset
      </button>
    </div>
    <div
      class="column column--m-3"
    >
      <button
        class="btn btn--primary btn--full-width"
        tabindex="0"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
